contracts/svm/programs/express_relay/src/utils.rs

```diff
@@ -109,47 +109,85 @@ pub fn get_matching_instructions(
 
 /// Extracts the bid paid from a SubmitBid instruction
 pub fn extract_bid_from_submit_bid_ix(submit_bid_ix: &Instruction) -> Result<u64> {
-    let submit_bid_args =
-        SubmitBidArgs::try_from_slice(&submit_bid_ix.data[8..]).map_err(|_| {
-            ProgramError::BorshIoError("Failed to deserialize SubmitBidArgs".to_string())
-        })?;
+    let submit_bid_args = SubmitBidArgs::try_from_slice(
+        &submit_bid_ix.data[crate::instruction::SubmitBid::DISCRIMINATOR.len()..],
+    )
+    .map_err(|_| ProgramError::BorshIoError("Failed to deserialize SubmitBidArgs".to_string()))?;
     Ok(submit_bid_args.bid_amount)
 }
 
-/// Performs instruction introspection on a transaction to determine the SubmitBid instructions that match the given permission and router
-/// Returns the number of matching instructions. If the permission_info struct is provided, also returns the total fees paid to the router in the matching instructions
+pub fn perform_fee_split_router(amount: u64, split_router: u64) -> Result<u64> {
+    let fee_router = amount * split_router / FEE_SPLIT_PRECISION;
+    if fee_router > amount {
+        // this error should never be reached due to fee split checks, but kept as a matter of defensive programming
+        return err!(ErrorCode::FeesHigherThanBid);
+    }
+    Ok(fee_router)
+}
+
+pub fn perform_fee_split_relayer(amount: u64, split_relayer: u64) -> Result<u64> {
```

Comment:
nit: these functions being public are dangerous since one would think you input the total bid amount and the splits and it would just work. But the amount input should be after the router fee deduction. we can make:
- make them private
- document them better
- inline their logic inside perform_fee_splits and use that in the cpi too